src/features/pass-survey/model/hooks/useActivityState.ts

```diff
@@ -171,43 +171,31 @@ function useActivityState({
     });
   }
 
-  function removeAnswer(step: number) {
-    const currentStorageRecord = getCurrentActivityStorageRecord();
-
-    if (!currentStorageRecord) {
-      return;
-    }
+  function removeConditionallyHiddenItemsAnswersAndTimers(fromStep: number) {
```

Comment:
I like the idea to combile callback and iteratePipeline, but 
1. let's keep the previous name - iteratePipeline or iterateWithConditionalLogic
2. it seems this snippet is not considered:
```
if (!currentStorageRecord?.timers) {
   return;
}
```

Reply:
2. yes, but not the same. In the prev. version it was just return, so {} wasn't recorder into the object unlike the current version. 
For example: !!undefined - false, !!{} - true.